src/network/networkctl.c

```diff
@@ -2919,6 +2928,270 @@ static int verb_reconfigure(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int get_arg_drop_in(char **ret) {
+        _cleanup_free_ char *drop_in = NULL;
+
+        assert(ret);
+
+        if (isempty(arg_drop_in))
+                return 0;
+
+        if (!endswith(arg_drop_in, ".conf"))
+                drop_in = strjoin(arg_drop_in, ".conf");
+        else
+                drop_in = strdup(arg_drop_in);
+        if (!drop_in)
+                return log_oom();
+
+        if (!filename_is_valid(drop_in))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid drop-in file name '%s'.", drop_in);
+
+        *ret = TAKE_PTR(drop_in);
+        return 1;
+}
+
+static int get_config_files_by_name(const char *name, char **ret_path, char ***ret_dropins) {
+        _cleanup_strv_free_ char **dropins = NULL, **dropin_dirs = NULL;
+        _cleanup_free_ char *path = NULL;
+        int r;
+
+        assert(name);
+        assert(ret_path);
+
+        STRV_FOREACH(i, NETWORK_DIRS) {
+                _cleanup_free_ char *p = NULL;
+
+                p = path_join(*i, name);
+                if (!p)
+                        return -ENOMEM;
+
+                r = strv_extendf(&dropin_dirs, "%s.d", p);
+                if (r < 0)
+                        return r;
+
+                if (!path && access(p, F_OK) >= 0)
```

Comment:
doesn't matter, but we usually try to deug log about unexpected errors from access(). i.e. if access() fails with any other error but ENOENT something is fishy so at the very least we should debug log about it